ros2sim/parsers/__init__.py

```diff
@@ -1,12 +1,152 @@
+    self.right_elbow, self.checksum, self.checksum_error, self.packet_available,
+    self.data_request)
+
+  def set_theta_value(self, s: str, val: int):
+    """Sets the imu theta value
+    """
+    if s == "θx":
+      self.theta_x = val
+    elif s == "θy":
+      self.theta_y = val
+    elif s == "θz":
+      self.theta_z = val
+    else:
+      raise ValueError("Imu axes name should be one of these: θx, θy, θz")
+  
+  def get_theta_value(self, s: str) -> int:
+    """Returns the imu value corresponding to the axes represented by s
+    """
+    if s == "θx":
+      return self.theta_x
+    elif s == "θy":
+      return self.theta_y
+    elif s == "θz":
+      return self.theta_z
+    else:
+      raise ValueError("Imu axes name should be one of these: θx, θy, θz")   
+
+  def set_joint_value_from_state(self, s: SerialReadState, value: int):
+    """Sets the joint value for the given SerialReadState
+    """
+    if s == SerialReadState.READ_L_HIP:
+      self.left_hip = value
+    elif s == SerialReadState.READ_L_KNEE:
+      self.left_knee = value
+    elif s == SerialReadState.READ_R_HIP:
+      self.right_hip = value
+    elif s == SerialReadState.READ_R_KNEE:
+      self.right_knee = value
+    elif s == SerialReadState.READ_L_SHOULDER:
+      self.left_shoulder = value
+    elif s == SerialReadState.READ_L_ELBOW:
+      self.left_elbow = value
+    elif s == SerialReadState.READ_R_SHOULDER:
+      self.right_shoulder = value
+    elif s == SerialReadState.READ_R_ELBOW:
+      self.right_elbow = value
+
+  def set_all_joint_values(self, left_hip=0, left_knee=0, right_hip=0, right_knee=0,
+    left_shoulder=0, left_elbow=0, right_shoulder=0, right_elbow=0):
+    """Sets the joint angles based on the values of the named arguments
+    """
+    self.left_hip = left_hip
+    self.left_knee = left_knee
+    self.right_hip = right_hip
+    self.right_knee = right_knee
+    self.left_shoulder = left_shoulder
+    self.left_elbow = left_elbow
+    self.right_shoulder = right_shoulder
+    self.right_elbow = right_elbow
+
+  def get_joint_values(self, ):
+    """Return a dictionary containing joint values based on joint ordering
+    """
+    pass
+
+  def get_joint_value_from_state(self, s: SerialReadState) -> int:
+    """Returns the joint value of the corresponding SerialReadState
+    """
+    if s == SerialReadState.READ_L_HIP:
+      return self.left_hip
+    elif s == SerialReadState.READ_L_KNEE:
+      return self.left_knee
+    elif s == SerialReadState.READ_R_HIP:
+      return self.right_hip
+    elif s == SerialReadState.READ_R_KNEE:
+      return self.right_knee
+    elif s == SerialReadState.READ_L_SHOULDER:
+      return self.left_shoulder
+    elif s == SerialReadState.READ_L_ELBOW:
+      return self.left_elbow
+    elif s == SerialReadState.READ_R_SHOULDER:
+      return self.right_shoulder
+    elif s == SerialReadState.READ_R_ELBOW:
+      return self.right_elbow
```

Comment:
Similar to the point above, this can become:

```suggestion
    for state, joint_value in self.state_map.items():
      if s == state:
        return joint_value
```